subsys/shell/modules/kernel_service.c

```diff
@@ -153,8 +172,8 @@ SHELL_STATIC_SUBCMD_SET_CREATE(sub_kernel,
 #if defined(CONFIG_REBOOT)
 	SHELL_CMD(reboot, &sub_kernel_reboot, "Reboot.", NULL),
 #endif
-#if defined(CONFIG_INIT_STACKS) && defined(CONFIG_THREAD_MONITOR) \
-				&& defined(CONFIG_THREAD_STACK_INFO)
+#if defined(CONFIG_INIT_STACKS) && defined(CONFIG_THREAD_STACK_INFO) && \
```

Comment:
just a hint - `SHELL_COND_CMD` macro could be utilized here, no potential dead code  since commands get compiled even when features are disabled.
```
#if defined(CONFIG_INIT_STACKS) && defined(CONFIG_THREAD_MONITOR) \
				&& defined(CONFIG_THREAD_STACK_INFO)
#define STACKS_AND_THREADS_CMDS 1
#endif

SHELL_STATIC_SUBCMD_SET_CREATE(sub_kernel,
	SHELL_CMD(cycles, NULL, "Kernel cycles.", cmd_kernel_cycles),
	SHELL_COND_CMD(CONFIG_REBOOT, reboot, &sub_kernel_reboot, "Reboot.", NULL),
	SHELL_COND_CMD(STACKS_AND_THREADS_CMDS, stacks, NULL,
			"List threads stack usage.", cmd_kernel_stacks),
	SHELL_COND_CMD(STACKS_AND_THREADS_CMDS, threads, NULL,
			"List kernel threads.", cmd_kernel_threads),
	SHELL_CMD(uptime, NULL, "Kernel uptime.", cmd_kernel_uptime),
	SHELL_CMD(version, NULL, "Kernel version.", cmd_kernel_version),
	SHELL_SUBCMD_SET_END /* Array terminated. */
);
```